src/typeguard/_checkers.py

```diff
@@ -209,12 +216,12 @@ def check_typed_dict(
     existing_keys = frozenset(value)
     extra_keys = existing_keys - declared_keys
     if extra_keys:
-        keys_formatted = ", ".join(f'"{key}"' for key in sorted(extra_keys))
+        keys_formatted = ", ".join(f'"{key}"' for key in _sorted_dict(extra_keys))
```

Comment:
I'm thinking that we don't need actually need to sort the keys. The order should be stable since Python 3.6. What if you dicth `_sorted_dict()` entirely?

Reply:
The order is actually lost just before, because frozensets are used to find the `extra_keys`.

Reply:
How is the order lost with frozensets?

Reply:
Never mind, yeah, they're not like dicts which keep key order.

Reply:
It does seem that frozensets already sort the keys, so I'm not sure why I was sorting the keys in the first place. Thoughts?

Reply:
So, this is confusing, but I think I have found the reason. :)
set / frozenset use the `hash` built-in function and sorts hashed values, but the `hash` function has a seed that changes when the interpreter is started. So when testing in an interactive shell, the order may be ordered alphabetically (that could occur randomly depending on the seed value), and this order is maintained when dealing with the same frozenset. But then, exiting the interactive shell and starting may yield a different ordering for the same values inside the frozenset.
This explains why in the unittest I had to put a regexp that matches both orders when the keys are not sorted.

Reply:
And dict keys maintain insertion order since Python 3.7 (https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects, see the note of the `iter` method)

Reply:
The hashing is described there: https://docs.python.org/3/reference/datamodel.html#object.__hash__

Reply:
Yeah, I commented earlier about dicts keeping insertion order (from Python 3.6 actually, but 3.7 made it official). But it seems that the sort order does not change over restarts of the interpreter, so the only purpose of the sorting seems to be to prettify the key list.

This is just for the looks, but would you please move the `_sorted_dict` function inside `check_typed_dict()`, as it's not used elsewhere? You can remove the leading underscore from the name too.

Reply:
Done.